### ISSUE TITLE
define-values evaluates its expression in a context where its names are shadowed

### ISSUE THREAD

**Issue by larceny-trac-import** [larcenists/larceny#681]
**Reported by: pnkfelix  on Wed Feb 20 06:32:08 2013**
```
> (let ((a 'hi))
    (define-values (a weird) (let ((w (not a))) (values '_ w)))
    weird)
#t
```

I would expect the above to evaluate to ```#f```, because in the scope of the expression for ```define-values``` above, ```a``` is bound to ```'hi```.

Admittedly, ```define-values``` is not in R5RS nor R6RS, and it is not 100% clear whether one might intend for recursive definitions to be valid, like this following:

```
(define-values (even? odd?)
  (values (lambda (n) (if (zero? n) #t (odd? (- n 1))))
          (lambda (n) (if (zero? n) #f (even? (- n 1))))))
```

Of course, it is easy to work around the issues here, regardless of whether ```define-values``` employs a shadowing or non-shadowing semantics.

In my opinion, it is more useful (and exposes less behind-the-scenes magic) to allow the former (non-shadowing) semantics, rather than the latter.

(To resolve a question like the above, rather than rely on my off-the-cuff opinion, it might be good to draft a SRFI for define-values.)

**Comment by WillClinger** [comment]
The R7RS definition of `define-values` will fix this.

**Comment by WillClinger** [comment]
Fixed by changeset 418ab11b93c280125d9b4d1de6f728222be9dff5
